rskj-core/src/main/java/org/ethereum/rpc/parameters/CallArgumentsParam.java

```diff
@@ -0,0 +1,93 @@
+package org.ethereum.rpc.parameters;
+
+import co.rsk.util.HexUtils;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import org.ethereum.rpc.CallArguments;
+import org.ethereum.rpc.exception.RskJsonRpcRequestException;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.math.BigInteger;
+
+@JsonDeserialize(using = CallArgumentsParam.Deserializer.class)
+public class CallArgumentsParam implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private transient final CallArguments callArguments;
+
+    public CallArgumentsParam(CallArguments callArguments) {
+        validateAddressProperty(callArguments.getFrom(), "from");
+        validateAddressProperty(callArguments.getTo(), "to");
+        validateNumberProperty(callArguments.getGas(), "gas");
+        validateNumberProperty(callArguments.getGasPrice(), "gasPrice");
+        validateNumberProperty(callArguments.getGasLimit(), "gasLimit");
+        validateNumberProperty(callArguments.getNonce(), "nonce");
+        validateNumberProperty(callArguments.getChainId(), "chainId");
+        validateNumberProperty(callArguments.getValue(), "value");
+        validateDataProperty(callArguments.getData());
+
+        this.callArguments = callArguments;
+    }
+
+    public CallArguments getCallArguments() {
+        return callArguments;
+    }
+
+    private boolean isPropertyValidHex(String propertyValue) {
```

Comment:
I think we should use existent parameters as properties instead of repeat validations along the different parameter classes.